sdk/storage/Azure.Storage.DataMovement/MigrationGuide.md

```diff
@@ -146,6 +146,153 @@ TODO
 
 ## Migration Samples
 
+This section contains side-by-side samples of legacy vs modern library usage of various features.
+These samples are not meant to be exhaustive, but demonstrate a wide variety of uses that may need migration.
+
+- Upload
+  - Upload single file to blob storage
+  - Upload directory to blob storage
+- Download
+  - Download single blob
+  - Download blob directory
+- Service to service copy
+  - Copy blob to blob
+  - Copy share file to blob
+
+### Upload
+
+#### Upload single file to blob storage
+
+**Legacy:**
+```csharp
+// these values provided by your code
+string filePath, containerName, blobName;
+CloudBlobClient client;
+```
+```csharp
+// upload blob
+await TransferManager.UploadAsync(
+    filePath,
+    client.GetContainerReference(containerName).GetBlockBlobReference(blobName));
+```
+**Modern:**
+```csharp
+// these values provided by your code
+string filePath, blobUri;
+LocalFilesStorageResourceProvider files;
+BlobsStorageResourceProvider blobs;
+TransferManager transferManager;
+```
+
+```csharp
+// upload blob
+TranferOperation operation = await transferManager.StartTransferAsync(
+    files.FromFile(filePath),
+    blobs.FromBlob(blobUri));
+await operation.WaitForCompletionAsync();
+```
+
+#### Upload directory to blob storage
+
+**Legacy:**
+```csharp
+// these values provided by your code
+string directoryPath, containerName, blobDirectoryPath;
+CloudBlobClient client;
+```
+```csharp
+// upload blob
+await TransferManager.UploadDirectoryAsync(
+    directoryPath,
+    client.GetContainerReference(containerName).GetDirectoryReference(blobDirectoryPath));
+```
+**Modern:**
+```csharp
+// these values provided by your code
+string directoryPath, containerUri, blobDirectoryPath;
+LocalFilesStorageResourceProvider files;
+BlobsStorageResourceProvider blobs;
+TransferManager transferManager;
+```
+
+```csharp
+// upload blobs
+TranferOperation operation = await transferManager.StartTransferAsync(
+    files.FromDirectory(directoryPath),
+    blobs.FromContainer(containerUri, new BlobStorageResourceContainerOptions()
+    {
+        BlobDirectoryPrefix = blobDirectoryPath,
+    }));
+await operation.WaitForCompletionAsync();
```

Comment:
In track 1, does calling UploadDirectoryAsync, actually wait the entire transfer?

Reply:
as far as i can tell, it gives you a task for transfer completion. I don't know this for certain yet, but other things also don't make sense to me if i'm wrong

Reply:
From what I could tell when writing perf tests, it does. There is no other way to wait